raphtory/src/io/parquet_loaders.rs

```
@@ -0,0 +1,341 @@
+    ),
+    GraphError,
+> {
+    let read_schema = |metadata: &FileMetaData| -> Result<ArrowSchema, GraphError> {
+        let schema = read::infer_schema(metadata)?;
+        let fields = schema
+            .fields
+            .iter()
+            .map(|f| {
+                if f.data_type == DataType::Utf8View {
+                    Field::new(f.name.clone(), DataType::LargeUtf8, f.is_nullable)
+                } else {
+                    f.clone()
+                }
+            })
+            .filter(|f| {
+                // Filtered fields to avoid loading data that is not needed
+                col_names.contains(&f.name.as_str())
+            })
+            .collect::<Vec<_>>();
+
+        Ok(ArrowSchema::from(fields).with_metadata(schema.metadata))
+    };
+
+    let mut file = std::fs::File::open(&path)?;
+    let metadata = read_metadata(&mut file)?;
+    let row_groups = metadata.clone().row_groups;
+    let schema = read_schema(&metadata)?;
+
+    // Although fields are already filtered by col_names, we need names in the order as it appears
+    // in the schema to create PretendDF
+    let names = schema.fields.iter().map(|f| f.name.clone()).collect_vec();
+
+    let reader = FileReader::new(file, row_groups, schema, None, None, None);
+    Ok((names, reader))
+}
+
+fn get_parquet_file_paths(parquet_path: &Path) -> Vec<PathBuf> {
+    let mut parquet_files = Vec::new();
+    if parquet_path.is_file() {
+        parquet_files.push(parquet_path.to_path_buf());
+    } else if parquet_path.is_dir() {
+        for entry in fs::read_dir(parquet_path).expect("Directory not found") {
+            let entry = entry.expect("Unable to read entry");
+            let path = entry.path();
+            if path.extension().map_or(false, |ext| ext == "parquet") {
+                parquet_files.push(path);
+            }
+        }
+    } else {
+        println!("Invalid path provided: {:?}", parquet_path);
```

This should throw an error no?